sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakePathClient.java

```diff
@@ -303,10 +404,48 @@ public Response<PathInfo> createWithResponse(String permissions, String umask, P
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Response<PathInfo> createWithResponse(DataLakePathCreateOptions options, Duration timeout, Context context) {
-        Mono<Response<PathInfo>> response = dataLakePathAsyncClient.createWithResponse(options, context);
-        return StorageImplUtils.blockWithOptionalTimeout(response, timeout);
-    }
+        DataLakePathCreateOptions finalOptions = options == null ? new DataLakePathCreateOptions() : options;
+        DataLakeRequestConditions requestConditions = options.getRequestConditions() == null ? new DataLakeRequestConditions() : options.getRequestConditions();
+
+        LeaseAccessConditions lac = new LeaseAccessConditions().setLeaseId(requestConditions.getLeaseId());
```

Comment:
Is there anything we could do to help share logic between the sync and async clients so when we make an update both are updated?

Reply:
Can try to address this in the follow-up PR